src/gpl/src/nesterovBase.h

```diff
@@ -1123,4 +1134,35 @@ class biNormalParameters
   float uy;
 };
 
+class GCellHandle
+{
+ public:
+  using StorageVariant = std::variant<NesterovBaseCommon*, NesterovBase*>;
+
+  GCellHandle(NesterovBaseCommon* nbc, size_t idx) : storage(nbc), index(idx) {}
+  GCellHandle(NesterovBase* nb, size_t idx) : storage(nb), index(idx) {}
+
+  // Non-const versions
+  GCell* operator->() { return &getGCell(); }
+  GCell& operator*() { return getGCell(); }
+  operator GCell*() { return &getGCell(); }
+
+  // Const versions
+  const GCell* operator->() const { return &getGCell(); }
+  const GCell& operator*() const { return getGCell(); }
+  operator const GCell*() const { return &getGCell(); }
+
+ private:
+  StorageVariant storage;
+  size_t index;
+
+  GCell& getGCell() const
+  {
+    if (std::holds_alternative<NesterovBaseCommon*>(storage)) {
+      return std::get<NesterovBaseCommon*>(storage)->getGCell(index);
+    }
+    return std::get<NesterovBase*>(storage)->getFillerGCell(index);
+  }
```

Comment:
Per https://google.github.io/styleguide/cppguide.html#Declaration_Order the order should be
```
using StorageVariant ...  // types
GCell& getGCell() const ... // then member funcs
  StorageVariant storage; // then data members
  size_t index;
```